app/data_client.go

```diff
@@ -235,26 +235,41 @@ type FileData struct {
 	Data []byte
 }
 
-// BinaryOptions represents optional parameters for the BinaryDataCaptureUpload method.
-type BinaryOptions struct {
+// DataByFilterOptions contains optional parameters for TabularDataByFilter and BinaryDataByFilter.
+type DataByFilterOptions struct {
+	// No Filter implies all data.
+	Filter *Filter
+	// Limit defaults to 50 if unspecified.
+	Limit int
```

Comment:
if Limit is not a pointer, how do we know whether they set the limit to 0 or whether they did not set it?

Reply:
A user would not put the limit to 0, or they would not get any data at all.

Reply:
The limit is there to limit how many entries to include in a page of data.